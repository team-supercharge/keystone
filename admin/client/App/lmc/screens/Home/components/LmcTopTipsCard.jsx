import React, { Component } from 'react';
import PropTypes from 'prop-types';

class LmcTopTipsCard extends Component {
    shouldComponentUpdate () {
        return false;
    }
    render () {
        const { video } = this.props;
        const defaultVideo = 'https://www.youtube.com/embed/sYB5GpaU6RE?rel=0&amp;controls=1&amp;showinfo=0';
        return (
            <div>
                <h2 className="lmc-card-title">
                    { TITLE }
                </h2>
                <div className="lmc-card" style={styles.card}>
                    {/* <div className="lmc-card-body"> */}
                        <div className="videoWrapper">
                            <iframe
                                style={styles.video}
                                ref="iframe"
                                src={video || defaultVideo}
                                frameBorder="0"
                                allowFullScreen />
                        </div>
                    {/* </div> */}
                    
                </div>
            </div>
            // <div>
            //     <h2 style={{ marginTop: 0 }} className="lmc-card-title">
            //         Top Tips
            //     </h2>
            // </div>
        );
    }
}
const styles = {
    card: {
        minHeight: 258,
        maxHeight: 260,
        borderRadius: 5,
        overflow: 'hidden',
    },
    video: {
        border: 'none', 
        maxHeight: 260
    }
};

const TITLE = 'Top Tips';

LmcTopTipsCard.propTypes = {
    video: PropTypes.string.isRequired,
};

export default LmcTopTipsCard;
