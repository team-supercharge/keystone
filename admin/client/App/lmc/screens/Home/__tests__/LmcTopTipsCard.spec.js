import React from 'react'
import { shallow } from 'enzyme'
import LmcTopTipsCard from '../components/LmcTopTipsCard'

describe('LmcTopTipsCard', () => {
    let wrapper
    let videoUrl

    beforeEach(() => {
        videoUrl = 'TestVideo'

        wrapper = shallow(
            <LmcTopTipsCard
                video={videoUrl}
            />
        )
    })

    test('should render correct snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('should render correct video', () => {
        const video = wrapper.find('iframe')
        expect(video.props().src).toEqual(videoUrl)
    })
})