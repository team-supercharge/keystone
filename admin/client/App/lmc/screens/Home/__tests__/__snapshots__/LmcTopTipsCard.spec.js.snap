// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcTopTipsCard should render correct snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h2
      className="lmc-card-title"
    >
      Top Tips
    </h2>
    <div
      className="lmc-card"
      style={
        Object {
          "minHeight": 258,
          "paddingTop": 18,
        }
      }
    >
      <div
        className="videoWrapper"
      >
        <iframe
          allowFullScreen={true}
          frameBorder="0"
          src="TestVideo"
          style={
            Object {
              "border": "none",
            }
          }
          width="345"
        />
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div>
      <h2
        className="lmc-card-title"
      >
        Top Tips
      </h2>
      <div
        className="lmc-card"
        style={
          Object {
            "minHeight": 258,
            "paddingTop": 18,
          }
        }
      >
        <div
          className="videoWrapper"
        >
          <iframe
            allowFullScreen={true}
            frameBorder="0"
            src="TestVideo"
            style={
              Object {
                "border": "none",
              }
            }
            width="345"
          />
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcTopTipsCard
        video="TestVideo"
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcTopTipsCard {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "video": "TestVideo",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h2
            className="lmc-card-title"
          >
            Top Tips
          </h2>
          <div
            className="lmc-card"
            style={
              Object {
                "minHeight": 258,
                "paddingTop": 18,
              }
            }
          >
            <div
              className="videoWrapper"
            >
              <iframe
                allowFullScreen={true}
                frameBorder="0"
                src="TestVideo"
                style={
                  Object {
                    "border": "none",
                  }
                }
                width="345"
              />
            </div>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h2
            className="lmc-card-title"
          >
            Top Tips
          </h2>
          <div
            className="lmc-card"
            style={
              Object {
                "minHeight": 258,
                "paddingTop": 18,
              }
            }
          >
            <div
              className="videoWrapper"
            >
              <iframe
                allowFullScreen={true}
                frameBorder="0"
                src="TestVideo"
                style={
                  Object {
                    "border": "none",
                  }
                }
                width="345"
              />
            </div>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcTopTipsCard
    video="TestVideo"
  />,
}
`;
